buffer/src/main/java/io/netty/buffer/DefaultByteBufHolder.java

```diff
@@ -43,12 +43,26 @@ public ByteBuf content() {
 
     @Override
     public ByteBufHolder copy() {
-        return new DefaultByteBufHolder(data.copy());
+        return copy(data.copy());
     }
 
     @Override
     public ByteBufHolder duplicate() {
-        return new DefaultByteBufHolder(data.duplicate());
+        return copy(data.duplicate());
+    }
+
+    @Override
+    public ByteBufHolder rduplicate() {
+        return copy(data.rduplicate());
+    }
+
+    /**
+     * Override this method to return a new copy of this object whose content is set to the specified {@code content}.
+     * The default implementation of {@link #copy()}, {@link #duplicate()} and {@link #rduplicate()} invokes this
+     * method to create a copy.
+     */
+    protected ByteBufHolder copy(ByteBuf content) {
```

Comment:
consider renaming `copy`. this nomenclature is overloaded w.r.t. `ByteBuf`'s meaning of copy.